.github/workflows/python-client-tests.yml

```diff
@@ -0,0 +1,45 @@
+name: Python Client Tests
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: '3.9'
+
+    - name: Generate Python client library
+      run: build/sbt generate
```

Comment:
do we need to generate all the clients for this?

Reply:
No, but I'm not sure what the command is to only generate the python client. What would that be?

Reply:
build/sbt pythonClient/generate

Reply:
can we move all these steps into a single python script like `clients/python/run-tests.py`?
so that its easy to manually run all the tests as well just by running this?

Reply:
Several of these steps (creating the virtual env + installing dependencies) should only be run once for setup, and may also depend on the configuration of python environments you have locally. Because of that, I don't think it makes sense to put them into a single script.

I've already documented the steps in the readme and I believe they should be fairly straightforward to run.